stonecrop/src/doctype.ts

```diff
@@ -1,17 +1,17 @@
 import { ComponentOptions } from 'vue'
 
-import { SchemaTypes } from '@agritheory/aform/types'
+import { Meta } from 'types/index'
 
 export default class Doctype {
 	doctype: string
-	schema: SchemaTypes[]
-	events: any // TODO: new Machine()
-	hooks: any
+	schema: Meta['schema']
+	events: Meta['events']
```

Comment:
I don't think this is right. These need to be immutable primitives. Meta as a type is duplication of the Doctype class / type.

Reply:
I can change the values to read-only, but did you want me to change the name as well?

```suggestion
	readonly events: Meta['events']
```

Reply:
I was thinking of something like this:

```
import { Map } from immutablejs
import { MachineConfig } from 'xstate'

schema:  Map
readonly events: MachineConfig
hooks: Map
```

`MachineConfig` isn't immutable [per the docs](https://xstate.js.org/docs/guides/typescript.html#config-objects), so we could add that later but it isn't in scope at this time.